### PR TITLE
Get physical CPU count from lscpu if available.

### LINKED ISSUE #60: New ps_cpu_count fails on non-x86 Linux
The new `ps_cpu_count` returns 0 on non-x86 systems ([aarch64](https://koji.fedoraproject.org/koji/taskinfo?taskID=32978623), [armv7hl](https://koji.fedoraproject.org/koji/taskinfo?taskID=32978626), [ppc64le](https://koji.fedoraproject.org/koji/taskinfo?taskID=32978622), and [s390x](https://koji.fedoraproject.org/koji/taskinfo?taskID=32978625)), causing the test to fail (as it expects non-zero result):
```
  == Failed ======================================================================
  -- 1. Failure: ps_cpu_count (@test-system.R#65)  -------------------------------
  phy > 0 isn't true.
```
It appears that `/proc/cpuinfo` is not too reliable. I don't have direct access to all these systems, but I can check ppc64le, which looks like this:
```
processor	: 0
cpu		: POWER8 (architected), altivec supported
clock		: 3425.000000MHz
revision	: 2.1 (pvr 004b 0201)

... other processors ...

timebase	: 512000000
platform	: pSeries
model		: IBM pSeries (emulated by qemu)
machine		: CHRP IBM pSeries (emulated by qemu)
MMU		: Hash
```
and armv7l, which looks like:
```
processor	: 0
model name	: ARMv7 Processor rev 0 (v7l)
BogoMIPS	: 2174.97
Features	: half thumb fastmult vfp edsp thumbee neon vfpv3 tls vfpd32 
CPU implementer	: 0x41
CPU architecture: 7
CPU variant	: 0x3
CPU part	: 0xc09
CPU revision	: 0

... other processors ...

Hardware	: Highbank
Revision	: 0000
Serial		: 01234567890123456789
```

### PR DESCRIPTION
**Pull request by QuLogic**
This appears to work for all systems I mentioned earlier: https://koji.fedoraproject.org/koji/taskinfo?taskID=33591831

Fixes #60.

### CODE REVIEW

File: R/linux.R [hunk 1]
```diff
@@ -140,7 +140,20 @@ psl__decode_address <- function(addr, family) {
...
-ps_cpu_count_physical_linux <- function() {
+psl__cpu_count_from_lscpu <- function() {
+  lines <- system("lscpu -p=core", intern = TRUE)
+  cores = list()
+
+  for (l in lines) {
+    if (!str_starts_with(l, "#")) {
+      cores[str_strip(l)] <- 1
+    }
+  }
```
Comment (QuLogic) [inline R/linux.R]: Not sure about the style here; this loop could maybe also be written as `cores <- unique(lines[!str_starts_with(lines, "#")])`

**Comment by QuLogic** [comment]
Build failure seems unrelated (`git2r` installation on macOS and some kind of progress thing on Windows.)

**Comment by QuLogic** [comment]
Rebased and the failure only seems to be about R 3.1, since some dependencies are no longer provided for it.

File: R/linux.R (hunk 1 above)
Comment (gaborcsardi) [reply R/linux.R]: Yeah, I'll change it to that.

**Comment by gaborcsardi** [comment]
Thanks much, and sorry for the delay!

**Pull request merged**